commands/config_e2e_test.go

```diff
@@ -1,14 +1,56 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
+
+// +build e2e
+
 package commands
 
 import (
+	"github.com/mattermost/mattermost-server/v5/model"
 	"github.com/spf13/cobra"
 
 	"github.com/mattermost/mmctl/client"
 	"github.com/mattermost/mmctl/printer"
 )
 
+func (s *MmctlE2ETestSuite) TestConfigResetCmdE2E() {
+	s.SetupTestHelper().InitBasic()
+
+	_, appErr := s.th.App.CreateUser(&model.User{Email: s.th.GenerateTestEmail(), Username: model.NewId(), Password: model.NewId()})
+	s.Require().Nil(appErr)
+
+	s.RunForSystemAdminAndLocal("System admin and local reset", func(c client.Client) {
+		printer.Clean()
+		s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.PrivacySettings.ShowEmailAddress = false })
+		defaultConfig := &model.Config{}
+		defaultConfig.SetDefaults()
```

Comment:
The config created here (`defaultConfig`) isn't linked to the server configs. So when we compare the values in the `Require` statements below, we aren't comparing with the config values of the server. 

We'll need to use the `GetConfig` method of the client to get the config from the server and then compare the returned value to the value we set above.

This applies to the test below as well

Reply:
the config is compared with s.th.App.Config() in line 29.

Reply:
>the config is compared with s.th.App.Config() in line 29.

I think you mean line 35?


I'm still not sure why lines 25, 26 and 32 are required?

Reply:
yeah..its 35,I was removing some lines in local, hence the confusion. Yeah, I think I can remove 25,26 and 32.

Reply:
I have made the changes.